packages/jellyfish-api-core/__tests__/category/evm/evmFeatureFlag.test.ts

```diff
@@ -0,0 +1,284 @@
+import { MasterNodeRegTestContainer } from '@defichain/testcontainers'
+import { ContainerAdapterClient } from '../../container_adapter_client'
+import { TransferDomainType } from '../../../src/category/account'
+import BigNumber from 'bignumber.js'
+
+describe('EVM feature flags', () => {
+  let dvmAddr: string, evmAddr: string, evmAddrTo: string
+  const container = new MasterNodeRegTestContainer()
+  const client = new ContainerAdapterClient(container)
+
+  async function dvmToEvmTransferDomain (): Promise<string> {
+    const txid = await client.account.transferDomain([
+      {
+        src: {
+          address: dvmAddr,
+          amount: '5@DFI',
+          domain: TransferDomainType.DVM
+        },
+        dst: {
+          address: evmAddr,
+          amount: '5@DFI',
+          domain: TransferDomainType.EVM
+        }
+      }
+    ])
+    return txid
+  }
+
+  async function evmToDvmTransferDomain (): Promise<string> {
+    const txid = await client.account.transferDomain([
+      {
+        src: {
+          address: evmAddr,
+          amount: '2@DFI',
+          domain: TransferDomainType.EVM
+
+        },
+        dst: {
+          address: dvmAddr,
+          amount: '2@DFI',
+          domain: TransferDomainType.DVM
+        }
+      }
+    ])
+    return txid
+  }
+
+  async function createEvmTx (nonce: number): Promise<string> {
+    const evmTxHash = await client.evm.evmtx({
+      from: evmAddr,
+      to: evmAddrTo,
+      value: new BigNumber(0.05),
+      gasPrice: 23,
+      gasLimit: 23000,
+      nonce
+    })
+    return evmTxHash
+  }
+
+  beforeAll(async () => {
+    await container.start()
+    await container.waitForWalletCoinbaseMaturity()
+
+    dvmAddr = await container.getNewAddress('dfiAddress', 'legacy')
+    evmAddr = await container.getNewAddress('eth', 'eth')
+    evmAddrTo = await container.getNewAddress('eth', 'eth')
+
+    await container.call('utxostoaccount', [{ [dvmAddr]: '100@0' }])
+    await container.generate(1)
+
+    await container.call('createtoken', [{
+      symbol: 'BTC',
+      name: 'BTC',
+      isDAT: true,
+      mintable: true,
+      tradeable: true,
+      collateralAddress: dvmAddr
+    }])
+    await container.generate(1)
+  })
+
+  afterAll(async () => {
+    await container.stop()
+  })
+
+  describe('all gov attrs for EVM is set to false', () => {
+    beforeAll(async () => {
+      await client.masternode.setGov({
+        ATTRIBUTES: {
+          'v0/params/feature/evm': 'false',
+          'v0/params/feature/transferdomain': 'false',
+          'v0/transferdomain/dvm-evm/enabled': 'false',
+          'v0/transferdomain/evm-dvm/enabled': 'false'
+        }
+      })
+      await container.generate(1)
+    })
+
+    it('should still successfully get eth address', async () => {
+      const newEvmAddr = await container.getNewAddress('eth', 'eth')
+      expect(newEvmAddr).toBeDefined()
+    })
+
+    it('should still successfully get eth balance', async () => {
```

Comment:
this test case is of no use as we are not getting any change in balance, I guess we can remove this

Reply:
i removed the dvm balance, but keeping the evm balance here just so we know includeEth is not affected by EVM flags at all

Reply:
in that case lets first transfer some token to evm and then only we will be able to check that diff